src/main/java/org/wildfly/security/auth/provider/ldap/UserPasswordCredentialLoader.java

```diff
@@ -71,43 +53,21 @@
 
     static final String DEFAULT_USER_PASSWORD_ATTRIBUTE_NAME = "userPassword";
 
-    public static final Set<String> SUPPORTED_ALGORITHMS = Collections.unmodifiableSet(
-        new HashSet<>(
-            Arrays.asList(ALGORITHM_CLEAR, ALGORITHM_SIMPLE_DIGEST_MD5, ALGORITHM_SIMPLE_DIGEST_SHA_1,
-                ALGORITHM_SIMPLE_DIGEST_SHA_256, ALGORITHM_SIMPLE_DIGEST_SHA_384, ALGORITHM_SIMPLE_DIGEST_SHA_512,
-                ALGORITHM_PASSWORD_SALT_DIGEST_MD5, ALGORITHM_PASSWORD_SALT_DIGEST_SHA_1, ALGORITHM_PASSWORD_SALT_DIGEST_SHA_256,
-                ALGORITHM_PASSWORD_SALT_DIGEST_SHA_384, ALGORITHM_PASSWORD_SALT_DIGEST_SHA_512, ALGORITHM_BSD_CRYPT_DES, ALGORITHM_CRYPT_DES)
-            )
-        );
-
     private final String userPasswordAttributeName;
 
-    // TODO - withCertaintly could be stores with the values in the map so some credential names are always supported and some only possible.
-    // Also control of which credentials can be persisted could have similar treatment.
-
-    private final boolean withCertainty;
-    private final Map<String, Set<String>> credentialNameAlgorithms;
-
-    UserPasswordCredentialLoader(final boolean withCertainty, final Map<String, Set<String>> credentialNameAlgorithms) {
```

Comment:
I think the administrator could easily restrict what credential types are supported just by policy.  Also, that statement would seem to apply to all realm types; maybe what we need is something on the front end of the realm which, in addition to e.g. auto-setting the multiple types of credential from a single credential, could also enforce this kind of restrictive policy on the types of credentials and algorithms that may be set.